scripts/check_harness_drift.py

```
@@ -341,10 +341,104 @@ def check_scripts_syntax(errors: list[str], verbose: bool) -> None:
+            if task_id and task_id in seen_tasks:
+                errors.append(
+                    f"AMCX-1: Duplicate task_id '{task_id}' in "
+                    f"{state_path.relative_to(ROOT)}"
+                )
+            if task_id:
+                seen_tasks[task_id] = True
+        if verbose and not any("AMCX-1" in e for e in errors):
+            print(f"[ok] AMCX-1: {state_path.relative_to(ROOT)} — {len(tasks)} tasks, clean")
+    except (json.JSONDecodeError, OSError) as exc:
+        errors.append(f"AMCX-1: Cannot read {state_path.relative_to(ROOT)}: {exc}")
+    if not amx_state_paths and not ecm_state_paths:
```

Ruff (0.16.1)

[error] Undefined name `amx_state_paths`

(F821)